src/DIRAC/Resources/IdProvider/Utilities.py

```diff
@@ -39,7 +39,8 @@ def getIdProviderIdentifierFromIssuerAndClientID(issuer, clientID):
 
     for identifier in result["Value"]:
         testedClientID = gConfig.getValue(f"/Resources/IdProviders/{identifier}/client_id")
-        if testedClientID and testedClientID == clientID:
+        # clientID is not always available, e.g. in case of a token of particular user
+        if not clientID or (testedClientID and testedClientID == clientID):
             # Found the client ID but need to check the issuer
```

Comment:
I am confused here: `clientID` might not be available, but then in the comment says "Found the client ID but..."?